### PR TITLE
Customizable initial application view loading from the config file

### LINKED ISSUE #773: Customize initial view
Opening to the rules view is a much better user experience as it does not immediately incur the expense of the trust load.

Thinking about it though, depending on the user's workflow, any of the views may be a good starting point.  It might be nice to allow customization of which view the app starts on.

One side effect of not starting with the trust admin view is that the initial system will not be checked.  Other views, like the analysis view, may need access to a checked trust database to provide accurate insight.  So the check still needs done somehow.

---
The following is probably a separate issue, but leaving it here for now.

What if the trust check for the initial system was a task separate from any view, that runs when the app opens.  It runs once regardless of what view the app starts on. Perhaps a single progress bar somewhere, like down in the status area could show the status of the background process.

From what I understand then the trust view would need to have progress bars, like it does now, that would show the progress of loading the trust into the table views.  These progress bars would only be indicating the status of the UI load, as the check was already completed.

The special case with that would be when the trust view is your opening view, there would be the disk check progress bar, plus the trust table progress bars.  How do they wire together?

It wouldn't even have to be the trust view is the starting view, could just be that the trust view is switched to prior to the check completing.

This relates to a few things that need reported
- Trust check continues after switching from view (UI)
- Trust check impacts focus on other views (UI)
- Trust check is repeated when switching back to trust view (UI)
- Trust check should be cancellable (Rust)

### LINKED ISSUE #804: Consistency in disabling tool menu items
When the application loads, the Trust tool is disabled.  That makes sense because trust is active.  However once switching away to a different tool the disable behavior is not applied to the new tool.

Just need to be consistent. Either remove the initial trust disabling, or make all views disabled when active.

### PR DESCRIPTION
**Pull request by dorschs57**
Right now this is manual process. To set your initial view update your config.toml file with a section like
```toml
[ui]
initial_view = 'trust'
```
The possible values are  "trust", "rules", "audit log", "syslog", or "profiler". With the default being "rules".

As part of this change the I removed the disabling of the trust menu option when you are on the trust screen. If the user is on the trust page and they hit the trust menu option again it just refreshes the page.

Closes #773 
Closes #804

### CODE REVIEW

**Review by jw3 — COMMENTED** [review]
Will need to add a `Requires:      python3-toml` to both spec files

**Comment by dorschs57** [comment]
@tparchambault Can you give the session saving and reloading a spin on this branch. I created a reusable widget to generate a file chooser dialog and switched all the places we were repeating ourselves in the code over to use it. The session management stuff used it in 3 different places.

**Comment by jw3** [comment]
@dorschs57 this picked up a conflict from master

**Comment by dorschs57** [comment]
@jw3 I put that minor change in to remove the + buttons.

File: fapolicy_analyzer/ui/features/system_feature.py [hunk 1]
```diff
@@ -185,7 +185,6 @@ def checking_finished():
             action_fn=ancillary_trust_load_complete,
             flag_fn=checking_finished,
         )
-        print(check_ancillary_trust)
```
Comment (jw3) [inline fapolicy_analyzer/ui/features/system_feature.py:188]: :+1:

File: fapolicy_analyzer/ui/types.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+from enum import Enum
+
+
+class PAGE_SELECTION(Enum):
+    TRUST_DATABASE_ADMIN = "trust"
+    RULES_ADMIN = "rules"
+    ANALYZE_FROM_AUDIT = "audit log"
```
Comment (jw3) [inline fapolicy_analyzer/ui/types.py:23]: This should be "debug log".

We have had some ambiguity between debug/audit in the past.  It doesnt need fixed now other than what is user facing here.

We updated the other user facing part in #771

**Review by jw3 — CHANGES_REQUESTED** [review]
Just the one comment on the /audit/debug/ replacement, otherwise :+1:

I tested all view configurations, including not present.  Also clicked through saving and loading sessions and a debug log.

File: fapolicy_analyzer/ui/types.py (hunk 2 above)
Comment (jw3) [reply fapolicy_analyzer/ui/types.py]: @dorschs57 I did a /audit/debug/ for this

**Comment by jw3** [comment]
@tparchambault any comment on this one?  I tested the pickers for the session operations, seemed to be in order.

File: fapolicy_analyzer/ui/types.py [hunk 3]
```diff
@@ -20,6 +20,6 @@
 class PAGE_SELECTION(Enum):
     TRUST_DATABASE_ADMIN = "trust"
     RULES_ADMIN = "rules"
-    ANALYZE_FROM_AUDIT = "audit log"
+    ANALYZE_FROM_DEBUG = "debug log"
```
Comment (dorschs57) [inline fapolicy_analyzer/ui/types.py:23]: 👍

**Review by dorschs57 — COMMENTED** [review]
@jw I'm good with your last commit. That was exactly what I had planned to do. If you tested it 👍

**Comment by tparchambault** [comment]
Currently reviewing it...

Can the generated toml file contain a comment with the possible valid choices for the option? Not this PR but downstream...

> The possible values are "trust", "rules", "audit log", "syslog", or "profiler".

**Comment by jw3** [comment]
> Can the generated toml file contain a comment with the possible valid choices for the option? Not this PR but downstream...

It could.  Though the comment in an existing config would be inaccurate if we add a new view.

User guide is probably the place for it. There is a section there now describing it.

**Comment by tparchambault** [comment]
Are we adding an additional `*.toml` file or renaming the existing `fapolicy-analyzer.toml` to `config.toml`? I'm hoping the latter.

**Comment by tparchambault** [comment]
Independent of our tool but something we need to be aware of, as something @jw3 also wrestled with yesterday:

> Mar 10 14:48:58 son-o-wimpy dbus-daemon[898]: [system] Successfully activated service 'org.fedoraproject.SetroubleshootPrivileged'
Mar 10 14:49:00 son-o-wimpy setroubleshoot[4401]: SELinux is preventing fapolicyd from write access on the file fapolicyd.pid. For complete SELinux messages run: sealert -l c719e1c0-b6e1-462a-b40f-ccbb8ba66612
Mar 10 14:49:00 son-o-wimpy setroubleshoot[4401]: SELinux is preventing fapolicyd from write access on the file fapolicyd.pid.
                                                  
Over RHEL8.6 via an RPM install, I was testing control of `fapolicyd` from `fapolicy-analyzer`.

**Comment by tparchambault** [comment]
Only one anomaly observed and I'm assuming it's independent of this PR: After `Deploying Changesets` and allowing the `Accept these changes?` count-down dialog to expire, which initiated an automatic Trust rollback, the previous pre-deployment session state was not reloaded or at least visible in the Ancillary Trust view, (which by itself is neither good nor bad, just a design choice), however the application state remained in the `pending changes` such that attempts to quit were met with the `You have applied changes` dialog. This may be a known issue.

**Review by tparchambault — COMMENTED** [review]
Tested over Rhel86. All good. No source comments that rise above the six of one, half dozen of another syntax and naming level. Redox and async is the future.

**Pull request merged**